src/doom/DoomMain.java

```diff
@@ -4074,482 +3878,140 @@ public boolean shouldPollLockingKeys() {
-        *  
-        *  Currently saves PCX screenshots, and only in devparm.
-        *  Very oldschool ;-)
-        *  
-        *  TODO: add non-devparm hotkey for screenshots, sequential screenshot
-        *  messages, option to save as either PCX or PNG. Also, request
-        *  current palette from VI (otherwise gamma settings and palette effects
-        *  don't show up).
-        *  
-        */
-
-       public void ScreenShot ()
-       {
-           int     i;
-           int[]  linear;
-           String format=new String("DOOM%d%d%d%d.png");
-           String lbmname = null;
-           
-           // munge planar buffer to linear
-           linear = (int[]) V.getScreen(DoomVideoRenderer.SCREEN_WS);
-           VI.ReadScreen (linear);
-
-           // find a file name to save it to
-           
-           int[] digit=new int[4];
-           
-           for (i=0 ; i<=9999 ; i++)
-           {
-           digit[0] = ((i/1000 )%10);
-           digit[1] =  ((i/100)%10);
-           digit[2] =  ((i/10)%10);
-           digit[3] =  (i%10);
-           lbmname=String.format(format, digit[0],digit[1],digit[2],digit[3]);
-           if (!C2JUtils.testReadAccess(lbmname))
-               break;  // file doesn't exist
-           }
-           if (i==10000)
-           I.Error ("M_ScreenShot: Couldn't create a PNG");
-
-           // save the pcx file
-           MenuMisc.WritePNGfile (lbmname, linear,
-                 SCREENWIDTH, SCREENHEIGHT);
-
-           players[consoleplayer].message = "screen shot";
-       }
+    @SuppressWarnings("unchecked")
+    private SceneRenderer<T, V> selectParallelRenderer2(int walls, int floors, int masked) {
+        switch (this.bppMode) {
+            case HiColor:
+                return (SceneRenderer<T, V>) new ParallelRenderer2
+                    .HiColor((DoomMain<byte[], short[]>) this, walls, floors, masked);
+            case TrueColor:
+                return (SceneRenderer<T, V>) new ParallelRenderer2
+                    .TrueColor((DoomMain<byte[], int[]>) this, walls, floors, masked);
+            case Indexed:
+            default:
+                return (SceneRenderer<T, V>) new ParallelRenderer2
+                    .Indexed((DoomMain<byte[], byte[]>) this, walls, floors, masked);
+        }
+    }
 
+    /**
+    *  M_Screenshot
+    *  
+    *  Currently saves PCX screenshots, and only in devparm.
+    *  Very oldschool ;-)
+    *  
+    *  TODO: add non-devparm hotkey for screenshots, sequential screenshot
+    *  messages, option to save as either PCX or PNG. Also, request
+    *  current palette from VI (otherwise gamma settings and palette effects
+    *  don't show up).
+    *  
+    */
+    public void ScreenShot() {
+        // find a file name to save it to
+        final String lbmname = findFileNameToSave(); // file doesn't exist
+
+        if (graphicSystem.writeScreenShot(lbmname, WS)) {
+            players[consoleplayer].message = "screen shot";
```

Comment:
This message must be loaded from the language files. If it's not there, then it must be added. This will be useful if we translate Mocha Doom. Right now there is English and French.